### DISCUSSION TITLE
RESTEasy Reactive using worker thread instead of IO thread

### DISCUSSION THREAD

**Discussion by tadenday** [quarkusio/quarkus#37289] [Community]
Hello,

I've encountered an issue while working with RESTEasy Reactive. I've created a simple endpoint as shown below:

```
@POST
@Path("/messages")
public Uni<Dto> create(@Valid Dto dto) {
    return service.createOrUpdate(Entity.from(dto));
}
```
When testing with small data, the task works well on the IO thread (vert.x-eventloop-thread-1). However, when testing with larger data, the task immediately switches to the worker thread (executor-thread-1).

I've attempted to force the task to be executed on the IO thread using the @NonBlocking annotation, but it hasn't resolved the issue.

I appreciate any assistance provided.

Best regards.

**Comment by mkouba** [comment]
> When testing with small data, the task works well on the IO thread (vert.x-eventloop-thread-1). However, when testing with larger data, the task immediately switches to the worker thread (executor-thread-1).

Hm, I believe that this should not happen. What exactly does `service.createOrUpdate()` do? Also what version of Quarkus do you use?
